tools/config_gen.py

```diff
@@ -164,30 +164,48 @@ def _print_options(opt_group, options):
         if opt.name in SKIP_OPTIONS:
             continue
 
+        opt_default = opt.default if opt.sample_default is None else opt.sample_default
+
         # Special case for options which could change during this script run
         static_option_value = STATIC_OPTION_VALUES.get(opt_group.name, {}).get(
             opt.name, None
         )
         if static_option_value:
-            opt.default = static_option_value
+            assert (
+                opt_default == static_option_value
+            ), f"opt_default={opt_default} != static_option_value={static_option_value}"
```

Comment:
Eventually, we can remove this `STATIC_OPTION_VALUES` stuff now that `olso_config` gives us a way to define the static option with the config using `sample_default`.